contracts/SP1_SpokePool.sol

```diff
@@ -0,0 +1,168 @@
+    /// stored by HubPool is relayed.
+    address public immutable hubPool;
+
+    /// @notice The address of the SP1 verifier contract.
+    address public verifier;
+
+    /// @notice The address of the Helios light client contract.
+    address public helios;
+
+    /// @notice The verification key for the acrossCall program.
+    bytes32 public acrossCallProgramVKey;
+
+    /// @notice Stores all proofs verified to prevent replay attacks.
+    mapping(bytes32 => bool) public verifiedProofs;
+
+    // Warning: this variable should _never_ be touched outside of this contract. It is intentionally set to be
+    // private. Leaving it set to true can permanently disable admin calls.
+    bool private _adminCallValidated;
+
+    event VerifiedProof(bytes32 indexed dataHash, address caller);
+
+    error NotHubPool();
+    error NotTarget();
+    error AdminCallAlreadySet();
+    error StateRootMismatch();
+    error AdminCallNotValidated();
+    error DelegateCallFailed();
+    error AlreadyReceived();
+
+    // All calls that have admin privileges must be fired from within the receiveL1State method that validates that
+    // the input data was published on L1 by the HubPool. This input data is then executed on this contract.
+    // This modifier sets the adminCallValidated variable so this condition can be checked in _requireAdminSender().
+    modifier validateInternalCalls() {
+        // Make sure adminCallValidated is set to True only once at beginning of the function, which prevents
+        // the function from being re-entered.
+        if (_adminCallValidated) {
+            revert AdminCallAlreadySet();
+        }
+
+        // This sets a variable indicating that we're now inside a validated call.
+        // Note: this is used by other methods to ensure that this call has been validated by this method and is not
+        // spoofed.
+        _adminCallValidated = true;
+
+        _;
+
+        // Reset adminCallValidated to false to disallow admin calls after this method exits.
+        _adminCallValidated = false;
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address _verifier,
+        address _helios,
+        bytes32 _acrossCallProgramVKey,
+        address _hubPool,
+        address _wrappedNativeTokenAddress,
+        uint32 _depositQuoteTimeBuffer,
+        uint32 _fillDeadlineBuffer
+    ) SpokePool(_wrappedNativeTokenAddress, _depositQuoteTimeBuffer, _fillDeadlineBuffer) {
+        verifier = _verifier;
+        helios = _helios;
+        acrossCallProgramVKey = _acrossCallProgramVKey;
+        hubPool = _hubPool;
+    }
+
+    /**
+     * @notice This can be called by an EOA to relay call data stored on the HubPool on L1 into this contract.
+     * @dev Consider making this an onlyOwner function so that only a priveleged EOA can relay state. This EOA
+     * wouldn't be able to tamper with the _publicValues.
```

Comment:
Just OOC, why do you think this should be privileged?

Reply:
I don't think it need to be but I'm flagging we should consider it